utils/utils.c

```diff
@@ -0,0 +1,75 @@
+#include "utils.h"
```

Comment:
Filename `utils/utils.c` is confusing. Can you use some functional names such `utils/url-rewriter.c`?